mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -98,32 +100,55 @@ void AffixTuningParameters::runOnFunction() {
   func.walk([&](miopen::GridwiseGemmOp op) {
     OpBuilder b(op.getContext());
 
-    // TBD. Compute tuning parameters from actual logic.
-    op.setAttr("block_size", b.getI32IntegerAttr(256));
+    ConvolutionContext convContext = populateConvContext(op);
+    InitParamsNonXDL validParams{0, 0, 0, 0, 0, 0};
+    GemmSize gemmSize;
+    DerivedParams gemmADerivedParam;
+    DerivedParams gemmBDerivedParam;
+    int64_t gemmCDstPerWrite;
+    int64_t gridSize;
 
-    op.setAttr("m_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("n_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("k_per_block", b.getI32IntegerAttr(8));
+    PopulateParams populateParams;
+    populateParams.paramsFromCtx(convContext, validParams, gemmSize,
+                                 gemmADerivedParam, gemmBDerivedParam,
+                                 gemmCDstPerWrite, gridSize);
 
-    op.setAttr("m_per_thread", b.getI32IntegerAttr(4));
-    op.setAttr("n_per_thread", b.getI32IntegerAttr(4));
-    op.setAttr("k_per_thread", b.getI32IntegerAttr(1));
+    // Tunable parameters.
+    op.setAttr("block_size", b.getI32IntegerAttr(validParams.blockSize));
```

Comment:
> Just a heads up I changed -miopen-affix-params a bit so it supplies block size and grid size to users via a callback. You need to do a rebase for this PR.

@whchung Wait, I already provided both gridsize and blocksize here. Why do we need this callback. If we do need to update block size and grid size, it should be populated through tuning parameters at its generation time, populating it in middle of the affix pass looks like a hack to me.

Reply:
I get myself confused for the last review. I think it is fine to override it from mlir-miopen-driver, but I need to update the tuning parameter logic to make sure it is coherent with the rest set of parameters. I will leave this as a topic for the next PR.